scripts/sct_dmri_moco.py

```diff
@@ -40,78 +41,106 @@ def get_parser():
+        add_help=None,
+        prog=os.path.basename(__file__).strip(".py")
+    )
+
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatory.add_argument(
+        '-i',
+        metavar=Metavar.file,
+        required=True,
+        help="Diffusion data. Example: dmri.nii.gz"
+    )
+    mandatory.add_argument(
+        '-bvec',
+        metavar=Metavar.file,
+        required=True,
+        help='Bvecs file. Example: bvecs.txt'
+    )
+
+    optional = parser.add_argument_group("\nOPTIONAL ARGUMENTS")
+    optional.add_argument(
+        "-h",
+        "--help",
+        action="help",
+        help="Show this help message and exit."
+    )
+    optional.add_argument(
+        '-bval',
+        metavar=Metavar.file,
+        help='Bvals file. Example: bvals.nii.gz',
+    )
+    optional.add_argument(
+        '-bvalmin',
+        type=float,
+        metavar=Metavar.float,
+        help='B-value threshold (in s/mm2) below which data is considered as b=0. Example: 50.0',
+    )
+    optional.add_argument(
+        '-g',
+        type=int,
+        metavar=Metavar.int,
+        help='Group nvols successive dMRI volumes for more robustness. Example: 2',
+        default=param_default.group_size,
+    )
+    optional.add_argument(
+        '-m',
+        metavar=Metavar.file,
+        help='Binary mask to limit voxels considered by the registration metric. Example: dmri_mask.nii.gz',
+    )
+    optional.add_argument(
+        '-param',
+        metavar=Metavar.str,
+        nargs='+',
+        help=f"R|Advanced parameters. Assign value with \"=\", and separate arguments with a space (\" \")\n"
+             f"    -poly [int]: Degree of polynomial function used for regularization along Z. For no regularization set to "
+             f"0. Default={param_default.poly}.\n"
+             f"    -smooth [mm]: Smoothing kernel. Default={param_default.smooth}.\n"
+             f"    -metric {{MI, MeanSquares, CC}}: Metric used for registration. Default={param_default.metric}.\n"
+             f"    -gradStep [float]: Searching step used by registration algorithm. The higher the more deformation "
+             f"allowed. Default={param_default.gradStep}.\n"
+             f"    -sample [None or 0-1]: Sampling rate used for registration metric. Default={param_default.sampling}.\n"
+    )
+    optional.add_argument(
+        '-x',
+        metavar=Metavar.str,
+        choices=['nn', 'linear', 'spline'],
+        default=param_default.interp,
+        help="Final interpolation."
+    )
+    optional.add_argument(
+        '-ofolder',
+        metavar=Metavar.folder,
+        default=param_default.path_out,
+        help="Output folder. Example: dmri_moco_results/"
+    )
+    # TODO: Convert -r to flag using action=store_true
+    optional.add_argument(
+        "-r",
+        metavar=Metavar.str,
+        choices=('0', '1'),
+        default='1',
+        help="Remove temporary files. 0 = no, 1 = yes"
+    )
```

Comment:
agreed about the store_true. Also agreed to deal with it in another PR. However, given that this change in CLI will provoke cross-compatibility issues, it would be nice to have it implemented before our planned 4.4.0 release.

Reply:
It may be better to leave argparse conversion (this PR and also scripts #11-20) for 4.5.0, then. I would have to complete the remaining scripts plus the two cross-compatibility changes in 1 week, which I think might be a little tight, time-wise. Some of those remaining scripts are quite a bit longer than the ones in this PR.

Would that be alright?

Reply:
we can also wait a bit more for 4.4.0

Reply:
Sounds good! I'll make this resolved for now and address this separately once all of the scripts have been converted. :)